packages/in_app_purchase/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -31,21 +32,28 @@ - (instancetype)initWithQueue:(nonnull SKPaymentQueue *)queue
                         updatedDownloads:(nullable UpdatedDownloads)updatedDownloads {
   self = [super init];
   if (self) {
-    self.queue = queue;
-    self.transactionsUpdated = transactionsUpdated;
-    self.transactionsRemoved = transactionsRemoved;
-    self.restoreTransactionFailed = restoreTransactionFailed;
-    self.paymentQueueRestoreCompletedTransactionsFinished = restoreCompletedTransactionsFinished;
-    self.shouldAddStorePayment = shouldAddStorePayment;
-    self.updatedDownloads = updatedDownloads;
-    self.transactionsSetter = [NSMutableDictionary new];
-    [queue addTransactionObserver:self];
+    _queue = queue;
+    _transactionsUpdated = transactionsUpdated;
+    _transactionsRemoved = transactionsRemoved;
+    _restoreTransactionFailed = restoreTransactionFailed;
+    _paymentQueueRestoreCompletedTransactionsFinished = restoreCompletedTransactionsFinished;
+    _shouldAddStorePayment = shouldAddStorePayment;
+    _updatedDownloads = updatedDownloads;
+    _transactionsSetter = [NSMutableDictionary dictionary];
   }
   return self;
 }
 
-- (void)addPayment:(SKPayment *)payment {
+- (void)startObservingPaymentQueue {
+  [_queue addTransactionObserver:self];
```

Comment:
We should find someway to indicate if the queue is observed. And all the other methods should throw if it is not observed. 
This can make users easier to debug if they forgot to observe.
Edit: After further review, I saw that this method is only called within the plugin, then it is less important to have the check. Might still have benefits for debugging and testing purpose. You call! :)

Reply:
Right now this method is called in the:
`- (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar`
I assume there won't be a case that `startObservingPaymentQueue ` is not called?

Another thing to mention, it's normally a bigger issue when this method is called multiple times during the app running such as user switches account during the app running, but I guess it might not be a concern for Flutter apps since plugin is only configured once when app launches?

Reply:
Does `+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar;` get called every time when a new registration happens? If so, I can just stop listening in this method.